packages/apollo/src/commands/service/list.ts

```diff
@@ -0,0 +1,173 @@
+import { table } from "heroku-cli-util";
+import moment from "moment";
+import { ApolloConfig, isServiceProject } from "apollo-language-server";
+import chalk from "chalk";
+import {
+  ListServices_service_implementingServices,
+  ListServices_service_implementingServices_FederatedImplementingServices_services
+} from "apollo-language-server/lib/graphqlTypes";
+
+interface TasksOutput {
+  config: ApolloConfig;
+  implementingServices: ListServices_service_implementingServices | null;
+}
+
+const formatImplementingService = (
+  implementingService: ListServices_service_implementingServices_FederatedImplementingServices_services
+) => {
+  return {
+    name: implementingService.name,
+    url: implementingService.url || "",
+    updatedAt: `${moment(implementingService.updatedAt).format(
+      "D MMMM YYYY"
+    )} (${moment(implementingService.updatedAt).fromNow()})`
+  };
+};
+
+function formatHumanReadable({
+  implementingServices,
+  graphName,
+  frontendUrl
+}: {
+  implementingServices: ListServices_service_implementingServices | null;
+  graphName: string | undefined;
+  frontendUrl: string | undefined;
+}): string {
+  let result = "";
+  if (
+    !implementingServices ||
+    implementingServices.__typename === "NonFederatedImplementingService"
+  ) {
+    result =
+      "\nThis graph is not federated, there are no services composing the graph";
+  } else if (implementingServices.services.length === 0) {
+    result = "\nThere are no services on this federated graph";
+  } else {
+    // Create a sorted list of the services.
+    const sortedImplementingServices = sortBy<
+      ListServices_service_implementingServices_FederatedImplementingServices_services
+    >(implementingServices.services, [service => service.name.toUpperCase()]);
+
+    table(
+      [...sortedImplementingServices.map(formatImplementingService)].filter(
+        Boolean
+      ),
+      {
+        columns: [
+          { key: "name", label: "Name" },
+          { key: "url", label: "Url" },
+          { key: "updatedAt", label: "Last Updated At" }
+        ],
+        // Override `printHeader` so we don't print a header
+        printHeader: () => {},
+        // The default `printLine` will output to the console; we want to capture the output so we can test
+        // it.
+        printLine: line => {
+          result += `\n${line}`;
+        }
+      }
+    );
+    const serviceListUrlEnding = `/service/${graphName}/service-list`;
+    const targetUrl = frontendUrl
+      ? `${frontendUrl}${serviceListUrlEnding}`
+      : `https://engine.apollographql.com${serviceListUrlEnding}`;
+    result += `\n\nView full details at: ${targetUrl}`;
+  }
+  return result;
+}
+
+export default class ServiceList extends ProjectCommand {
+  static description = "List the services in a graph";
+  static flags = {
+    ...ProjectCommand.flags,
+    tag: flags.string({
+      char: "t",
+      description: "The published tag to list the services from"
```

Comment:
nit: variant? IDK what our plan is for that migration tbh, but it's confusing that we're using graph everywhere and still using tag.

Reply:
Agreed, but the whole cli would have to make this change before this PR does!